filebeat/input/journald/input.go

```diff
@@ -207,54 +215,12 @@ func initCheckpoint(log *logp.Logger, c cursor.Cursor) checkpoint {
 	return cp
 }
 
-func withFilters(filters journalfield.IncludeMatches) func(*sdjournal.Journal) error {
-	return func(j *sdjournal.Journal) error {
-		return journalfield.ApplyIncludeMatches(j, filters)
-	}
-}
-
-func withUnits(units []string) func(*sdjournal.Journal) error {
-	return func(j *sdjournal.Journal) error {
-		return journalfield.ApplyUnitMatchers(j, units)
-	}
-}
-
-func withTransports(transports []string) func(*sdjournal.Journal) error {
-	return func(j *sdjournal.Journal) error {
-		return journalfield.ApplyTransportMatcher(j, transports)
-	}
-}
-
-func withSyslogIdentifiers(identifiers []string) func(*sdjournal.Journal) error {
-	return func(j *sdjournal.Journal) error {
-		return journalfield.ApplySyslogIdentifierMatcher(j, identifiers)
-	}
-}
-
-// seekBy tries to find the last known position in the journal, so we can continue collecting
-// from the last known position.
-// The checkpoint is ignored if the user has configured the input to always
-// seek to the head/tail/since of the journal on startup.
-func seekBy(log *logp.Logger, cp checkpoint, seek, defaultSeek journalread.SeekMode) (mode journalread.SeekMode, pos string) {
-	mode = seek
-	if mode == journalread.SeekCursor && cp.Position == "" {
-		mode = defaultSeek
-		switch mode {
-		case journalread.SeekHead, journalread.SeekTail, journalread.SeekSince:
-		default:
-			log.Error("Invalid option for cursor_seek_fallback")
-			mode = journalread.SeekHead
-		}
-	}
-	return mode, cp.Position
-}
-
 // readerAdapter wraps journalread.Reader and adds two functionalities:
 //   - Allows it to behave like a reader.Reader
 //   - Translates the fields names from the journald format to something
 //     more human friendly
 type readerAdapter struct {
-	r                  *journalread.Reader
+	r                  journalReader
```

Comment:
If `Next` returns an error and we are temporarily not able to read, that would be a good place to set the input status to degraded. Similar for any other errors that are not related to specific events and result in not reading from the journal. Event related errors are probably better to record with input metrics.

Reply:
I believe the input `Run` method is the best place for it as it is where the loop calling `Next` is. As I mentioned on https://github.com/elastic/beats/pull/40061#discussion_r1669361822, I believe status reporting should be a separated PR, I'll create an issue and link it in the Jourald epic.

Reply:
Here is the issue: https://github.com/elastic/beats/issues/40143, it's already linked as required to GA the journald input.